sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -1519,6 +1519,19 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
             |ORDER BY sum(b) + 1
           """.stripMargin),
       Row("4", 3) :: Row("1", 7) :: Row("3", 11) :: Row("2", 15) :: Nil)
+
+    Seq("1" -> 3, "2" -> 7, "2" -> 8, "3" -> 5, "3" -> 6, "3" -> 2, "4" -> 1, "4" -> 2,
+      "4" -> 3, "4" -> 4).toDF("a", "b").registerTempTable("orderByData2")
```

Comment:
For `orderByData` the group by result would be ("4", 2), ("3", 2), ("2", 2), ("1", 2), so order by count(*) would be ambiguous.